packages/react/src/auth/useSignInAnonymouslyMutation.test.tsx

```diff
@@ -1,29 +1,53 @@
 import React from "react";
-import { describe, expect, test, beforeEach, vi } from "vitest";
+import {
+  describe,
+  expect,
+  test,
+  beforeEach,
+  vi,
+  type MockInstance,
+} from "vitest";
 import { renderHook, act, waitFor } from "@testing-library/react";
 import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
-import { type Auth, type UserCredential } from "firebase/auth";
+import {
+  type Auth,
+  type UserCredential,
+  signInAnonymously,
+} from "firebase/auth";
 import { useSignInAnonymouslyMutation } from "./useSignInAnonymouslyMutation";
 
+vi.mock("firebase/auth", () => ({
```

Comment:
@Ehesp i did have 2 options here when mocking the `firebase auth module` 
1. Option One (seemed simple and its what i went with)
```tsx
// this ignores all other imports from the module 
// and only mocks the specific function 'signInAnonymously'
vi.mock("firebase/auth", () => ({
  signInAnonymously: vi.fn(),
}));
```
3. Option Two
```tsx
//
vi.mock("firebase/auth", async (importOriginal) => {
  // Import the original 'firebase/auth' module
  const module = await importOriginal<typeof import("firebase/auth")>();

  // Return a mocked version of the module
  return {
    ...module, // Spread in all the original functions from the module
    signInAnonymously: vi.fn(), // Mock the specific function 'signInAnonymously'
  };
});

```


do you think it would be better to with option 2 that returns a mocked version of the module with all original functions ?